src/peepo_drum_kit/chart_editor_i18n.cpp

```diff
@@ -0,0 +1,224 @@
+#include "chart_editor_i18n.h"
+
+namespace PeepoDrumKit::i18n
+{
+	static std::map<u32, std::string> HashStringMap;
+	static std::shared_mutex HashStringMapMutex;
+	std::string SelectedFontName = "NotoSansCJKjp-Regular.otf";
+	std::vector<LocaleEntry> LocaleEntries;
+
+	static void InitBuiltinLocaleWithoutLock()
+	{
+		FontMainFileNameTarget = FontMainFileNameDefault;
+		HashStringMap.clear();
+
+#define X(en, jp) HashStringMap[Hash(en)] = std::string(en);
+		PEEPODRUMKIT_UI_STRINGS_X_MACRO_LIST_JA
+#undef X
+	}
+
+	void InitBuiltinLocale()
+	{
+		HashStringMapMutex.lock();
+		InitBuiltinLocaleWithoutLock();
+		HashStringMapMutex.unlock();
+	}
+
+	cstr HashToString(u32 inHash)
+	{
+		HashStringMapMutex.lock_shared();
+		// if (HashStringMap.empty()) InitBuiltinLocale();
+		for (auto it = HashStringMap.begin(); it != HashStringMap.end(); ++it)
+			if (it->first == inHash)
+			{
+				cstr result = it->second.c_str();
+				HashStringMapMutex.unlock_shared();
+				return result;
+			}
+		HashStringMapMutex.unlock_shared();
```

Comment:
The map is already a hash table.
```suggestion
		{
			HashStringMapMutex.lock_shared();
			defer { HashStringMapMutex.unlock_shared(); };
			// if (HashStringMap.empty()) InitBuiltinLocale();
			auto it = HashStringMap.find(inHash);
			if (it != HashStringMap.end()) {
				cstr result = it->second.c_str();
				return result;
			}
		}
```

Reply:
Oops, hash table is `std::unordered_map`. `std::map` is binary tree.

Reply:
> Oops, hash table is `std::unordered_map`. `std::map` is binary tree.

fixed